### PR TITLE
Address PR comments on return omission.

### PR DESCRIPTION
**Pull request by nate-chandler**
Addresses two issues identified by Slava in the already merged return omission PR.

### CODE REVIEW

**Comment by nate-chandler** [comment]
@swift-ci please smoke test

**Comment by nate-chandler** [comment]
@swift-ci please smoke test

File: lib/Parse/ParseDecl.cpp [hunk 1]
```diff
@@ -5694,9 +5694,7 @@ void Parser::parseAbstractFunctionBody(AbstractFunctionDecl *AFD) {
     if (!Body.hasCodeCompletion() && BS->getNumElements() == 1) {
       auto Element = BS->getElement(0);
       if (auto *stmt = Element.dyn_cast<Stmt *>()) {
-        auto kind = AFD->getKind();
-        if (kind == DeclKind::Var || kind == DeclKind::Subscript ||
-            kind == DeclKind::Func ) {
+        if (isa<AccessorDecl>(AFD) || isa<FuncDecl>(AFD)) {
```
Comment (slavapestov) [inline lib/Parse/ParseDecl.cpp]: AccessorDecl is a subclass of FuncDecl so only the RHS of the `||` is needed.

**Review by slavapestov — APPROVED** [review]
Thank you! Only one minor nitpick if you allow me

File: lib/Parse/ParseDecl.cpp (hunk 1 above)
Comment (nate-chandler) [reply lib/Parse/ParseDecl.cpp]: Fixed!

**Comment by nate-chandler** [comment]
@swift-ci please smoke test

**Pull request merged**